### PR TITLE
ALTAPPS-544: Shared Sentry transactions convenient API

### PR DESCRIPTION
**Pull request by XanderZhu**
**YouTrack Issues**:
[ALTAPPS-544](https://vyahhi.myjetbrains.com/youtrack/issue/ALTAPPS-544)

**Checklist**

_Before Code Review:_

- [x] Fields "Assignees, Labels, Milestone" are filled in the pull request;
- [x] All checks have been passed;
- [x] Changes have been checked locally.

**Description**
- Implement `SentryInteractor.withTransaction` extension to automatically start and finish a transaction

### CODE REVIEW

**Comment by XanderZhu** [comment]
@ivan-magda @vladkash 
I've updated `ProjectSelectionListActionDispatcher.handleFetchContentAction`, `HomeActionDispatcher.fetchHomeScreenData` and `ProgressScreenActionDispatcher.handleFetchTrackWithProgressAction` to show the new API in the action.
Please, provide your feedback: how do you like it.

File: shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeActionDispatcher.kt [hunk 1]
```diff
@@ -137,6 +107,30 @@ class HomeActionDispatcher(
         }
     }
 
+    private suspend fun fetchHomeScreenData() {
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/home/presentation/HomeActionDispatcher.kt:110]: Let's use the same style for action handlers helper functions - if necessary we provide action and onNewMessage block.

See for example `ProgressScreenActionDispatcher.handleFetchTrackWithProgressAction` and `ProjectSelectionListActionDispatcher.handleFetchContentAction`

**Review by ivan-magda — APPROVED** [review]
> Please, provide your feedback: how do you like it.

Looks nice

File: shared/src/commonMain/kotlin/org/hyperskill/app/progress_screen/presentation/ProgressScreenActionDispatcher.kt [hunk 2]
```diff
@@ -50,59 +51,39 @@ internal class ProgressScreenActionDispatcher(
...
-            onNewMessage(
                 ProgressScreenFeature.TrackWithProgressFetchResult.Success(
                     trackWithProgress = trackWithProgress,
                     studyPlan = studyPlan,
                     profile = profile
                 )
-            )
-        }
+            }
+        }.let(onNewMessage)
```
Comment (ivan-magda) [inline shared/src/commonMain/kotlin/org/hyperskill/app/progress_screen/presentation/ProgressScreenActionDispatcher.kt:86]: Just a minor question:

Personally for me, the use of `let` means that we want to execute a code block containing a non-null value with a safe call operator `?` call before.

What's preferable to use here `let` or `also` scope function?

Comment (XanderZhu) [reply shared/src/commonMain/kotlin/org/hyperskill/app/progress_screen/presentation/ProgressScreenActionDispatcher.kt:86]: In this case, there isn't a big difference in which scope function to use.
But, in my opinion, `let` is preferable according to the name of the function.

**Comment by vladkash** [comment]
Looks good to me 👍

**Review by vladkash — APPROVED** [review]

**Pull request merged**